File: standalone/dzen2-hlwm/haskell/panel.hs
-- This is a modularized config for herbstluftwm tags in dzen2 statusbar

import System.Environment
import System.Process

import MyHelper
import MyPipeHandler

-- initialize

panelHeight = 24

-- ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- -----
-- main

main = do
    -- initialize
    
    args <- getArgs
    let monitor = getMonitor args
        
    geometry <- getGeometry monitor

    system "pkill dzen2"
    system $ "herbstclient pad " ++ show(monitor) ++ " "
        ++ show(panelHeight) ++ " 0 " ++ show(panelHeight) ++ " 0"

    -- run process in the background

    let paramsTop = getParamsTop panelHeight geometry
    detachDzen2 monitor paramsTop

    let paramsBottom = getParamsBottom panelHeight geometry
    detachDzen2Conky paramsBottom

    -- optional transparency
    detachTransset

    -- end of IO
    return ()

File: standalone/dzen2-hlwm/lua/panel.lua
#!/usr/bin/lua
-- This is a modularized config for herbstluftwm tags in dzen2 statusbar

local dirname  = debug.getinfo(1).source:match("@?(.*/)")
package.path   = package.path .. ';' .. dirname .. '?.lua;'

local helper      = require('.helper')
local pipehandler = require('.pipehandler')

-- ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- -----
-- main

panel_height = 24
monitor = helper.get_monitor(arg)

os.execute('pkill dzen2')
os.execute('herbstclient pad ' .. monitor .. ' ' 
    .. panel_height .. ' 0 ' .. panel_height .. ' 0')

-- run process in the background

local params_top = helper.get_params_top(monitor, panel_height)
pipehandler.detach_dzen2(monitor, params_top)

local params_bottom = helper.get_params_bottom(monitor, panel_height)
pipehandler.detach_dzen2_conky(params_bottom)

-- optional transparency
pipehandler.detach_transset()
